imap_processing/codice/utils.py

```diff
@@ -111,8 +112,11 @@ def create_dataset(packets: list[space_packet_parser.parser.Packet]) -> xr.Datas
     for packet in packets:
         add_metadata_to_array(packet, metadata_arrays)
 
+    # Convert to datetime64 and normalize by launch date
+    epoch_times = [calc_start_time(item) for item in metadata_arrays["SHCOARSE"]]
+
     epoch_time = xr.DataArray(
-        metadata_arrays["SHCOARSE"],
+        epoch_times,
         name="Epoch",
         dims=["Epoch"],
         attrs=ConstantCoordinates.EPOCH,
```

Comment:
If you like to keep raw SHCOARSE data, you will need to remove if statement from below for loop(line 131 in this file)

Reply:
Good catch, thank you!